client/patch-react.js

```diff
@@ -3,6 +3,15 @@
 
 import React from 'react'
 
+const getWritableProto = (obj, p) => {
+  const descr = Object.getOwnPropertyDescriptor(obj, p)
+  if (descr) {
+    return descr.writable ? obj : null
+  }
+  const proto = Object.getPrototypeOf(obj)
+  return proto ? getWritableProto(proto, p) : null
```

Comment:
Could we add a comment in the code to explain the motivation for this? Since it's a bit obscure